bundles/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/internal/discovery/NukiDeviceDiscoveryService.java

```diff
@@ -69,14 +70,22 @@ private DiscoveryResult createDiscoveryResult(BridgeApiListDeviceDto device, Nuk
                 .withRepresentationProperty(NukiBindingConstants.PROPERTY_NUKI_ID)
                 .withProperty(NukiBindingConstants.PROPERTY_NAME, device.getName())
                 .withProperty(NukiBindingConstants.PROPERTY_NUKI_ID, device.getNukiId())
+                .withProperty(NukiBindingConstants.PROPERTY_DEVICE_TYPE, device.getDeviceType())
                 .withProperty(NukiBindingConstants.PROPERTY_FIRMWARE_VERSION, device.getFirmwareVersion()).build();
     }
 
     private ThingUID getUid(String nukiId, int deviceType, NukiBridgeHandler bridgeHandler) {
```

Comment:
Please add `throw IllegalArgumentException` on this method and handle the exception in the method `createDiscoveryResult`

Reply:
I'm not sure I follow. Where should I put the throw and why?

Reply:
`private ThingUID getUid(String nukiId, int deviceType, NukiBridgeHandler bridgeHandler) throw IllegalArgumentException  {`

Catch this exception in method createDiscoveryResult which is calling the method getUid.

Reply:
Right I get it now. That case can "theoretically" never happen, since the device list is first filtered and only devices with supported types are passed to `createDiscoveryResult`, see
```
                        .filter(device -> NukiBindingConstants.DEVICE_DISCOVERY_SUPPORTED_DEVICES
                                .contains(device.getDeviceType()))
```
But thinking about it it's not really nice to have that check in two places, I'll drop the filter.

Reply:
I've removed the exception since I want to keep the stream and exception handling does not play well with it. Is this ok?

Reply:
Ok for me.